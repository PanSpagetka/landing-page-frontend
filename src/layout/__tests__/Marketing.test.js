/*global describe, test, expect */

import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { mapStateToProps } from '../Marketing';
import ConnectedMarketing, { Marketing } from '../Marketing';
import { ChartSpikeIcon } from '@patternfly/react-icons';

const marketingTechnologiesMock = [
    {
        title: 'title',
        id: 'id',
        marketingImage: ChartSpikeIcon,
        marketingText: 'marketingText',
        marketingUrls: {
            learnMore: 'https://www.redhat.com/en/technologies/management/insights'
        }
    }
];

const mockStore = configureMockStore();
const store = mockStore(marketingTechnologiesMock);

function getInput(obj) {
    return {
        technologies: {
            marketingTechnologies: [ obj ]
        }
    };
}

function getOutput(obj) {
    return { technologies: [ obj ]};
}

describe('mapStateToProps', () => {
    test('should return the re-wrapped data', () => {
        const data = { foo: 'bar' };
        const output = mapStateToProps(getInput(data));
        expect(output).toEqual(getOutput(data));
    });
});

describe('render Marketing component', () => {
    it('should render correctly connected', () => {
        const wrapper = mount(
            <Provider store={ store }>
                <ConnectedMarketing/>
            </Provider>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render correctly', () => {
        const wrapper = shallow(
            <Marketing/>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
