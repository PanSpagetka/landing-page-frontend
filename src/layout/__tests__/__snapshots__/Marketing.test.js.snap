// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Marketing component should render correctly 1`] = `
<Fragment>
  <Hero
    className="ins-p-marketing__hero"
    needsCTA={true}
    title="Explore our open, multicloud tools"
  />
  <PageSection
    className="ins-c-marketing pf-m-no-fill"
  >
    <Grid
      gutter="sm"
      md={6}
      sm={12}
      xl={6}
      xl2={3}
    />
  </PageSection>
</Fragment>
`;
